services/m3aggregator/main/main.go

```diff
@@ -121,6 +121,7 @@ func main() {
 
 	select {
 	case <-closedCh:
+		aggregator.Close()
```

Comment:
Shouldn't this happen in the goroutine just before `close(closedCh)`?  Otherwise it could block and it my surpass the gracefulShutdownTimeout